pkg/installmanager/installmanager.go

```diff
@@ -351,6 +351,12 @@ func (m *InstallManager) Run() error {
 		m.log.WithError(err).Error("error while trying to preemptively clean up")
 		return err
 	}
+	// Fail the cluster provision if infraID is set. This can occur if previous install
+	// pod for current cluster provision is unexpectedly killed or evicted.
+	if provision.Spec.InfraID != nil {
+		m.log.WithError(err).Error("infraID is already set on the ClusterProvision. Unexpected install pod restart detected")
```

Comment:
I can't tell where this `err` came from. I suspect you just meant to log your message. But I'm not sure you need to do that, as the controller-runtime will log the error (below) anyway. TL;DR: you can probably just remove this line.

Reply:
nice catch! I will just remove this line.

Reply:
```
time="2022-01-17T19:36:01Z" level=info msg="deleting existing object" installID=xjlzmk2s object=default/arane-05-0-4v524-admin-kubeconfig
time="2022-01-17T19:36:01Z" level=info msg="deleting existing object" installID=xjlzmk2s object=default/arane-05-0-4v524-admin-password
time="2022-01-17T19:36:01Z" level=error msg="infraID is already set on the ClusterProvision. Unexpected install pod restart detected" error="<nil>" installID=xjlzmk2s
time="2022-01-17T19:36:01Z" level=fatal msg="runtime error" error="infraID is already set on the ClusterProvision. Unexpected install pod restart detected"
```

Not controller runtime but returned error will be logged by [this](https://github.com/openshift/hive/blob/588f139c6122c62fd5232e7b47125bad103680fe/pkg/installmanager/installmanager.go#L171-L173) line of code